Always get attr static out

voznesenskym opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #95771

Discussion here https://github.com/pytorch/pytorch/issues/95630#issuecomment-1449596766

cc @soumith @yanboliang @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @desertfire

voznesenskym committed:
[RFC] Always get attr static out?

[ghstack-poisoned]

voznesenskym committed:
Update on "[RFC] Always get attr static out?"


Discussion here https://github.com/pytorch/pytorch/issues/95630#issuecomment-1449596766

cc soumith yanboliang penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx desertfire

[ghstack-poisoned]

ezyang commented on torch/_dynamo/variables/builder.py:
```diff
@@ -320,6 +320,10 @@ def _wrap(self, value):
 
+        # Note - There are some nested values where types mismatch!
+        # We want to get those out and wrap those.
+        value = inspect.getattr_static(value, "_torchdynamo_inline", value)
```
Don't you want to do this before the type()/id() matches then?

voznesenskym replied:
I was not sure, actually. Reading the code - it does not matter today, but maybe it will? I'll move it up.

voznesenskym replied:
Actually, I think here is better, this can only happen on types that are not subject to the above.

ezyang commented:
I don't really understand the semantics of `_torchdynamo_inline`, but I am willing to give this a shot.

jansel commented:
> I don't really understand the semantics of `_torchdynamo_inline`, but I am willing to give this a shot.

If you have:
```
@torch.compile
def foo(): ...


@torch.compile
def bar():
   return foo()
```

Then `foo._torchdynamo_inline` will point to the original `foo` (before the decorator wrapped it up in a context manager).  This allows inlining dynamo decorated things.

voznesenskym commented:
@pytorchbot merge

voznesenskym committed:
Update on "Always get attr static out"


Discussion here https://github.com/pytorch/pytorch/issues/95630#issuecomment-1449596766

cc soumith yanboliang penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx desertfire

[ghstack-poisoned]

voznesenskym commented:
@pytorchbot merge -f "Passed CI, rebase is safe"

atalman commented:
cc @voznesenskym @ngimel 

This issue is in the  milestones :[ https://github.com/pytorch/pytorch/milestone/36?closed=1](https://github.com/pytorch/pytorch/milestone/36?closed=1), if you want to see your fix included in this minor release. Please post it as a cherry-pick into the [[v2.0.1] Release Tracker](https://github.com/pytorch/pytorch/issues/97272). 

**The deadline is April 14, 5PM PST.**

**Only issues that have ‘cherry-picks’ will be considered for the release.**

Common FAQs:

Q1: Where can I find more information on the release process and terminology?

A: [pytorch/RELEASE.md at master · pytorch/pytorch · GitHub](https://github.com/pytorch/pytorch/blob/master/RELEASE.md)

Q2: Am I guaranteed to be included in the cherry-pick if I do above?

A: No, it is not guaranteed, the Release Team will review all submissions against the listed criteria before making the final decision on what to include on 4/17.

Q3: When is 2.1 going to be released?

A: We do not have a formal date at this time but will update the community when we do. Our immediate focus is 2.0.1. Note that 1.12 was released on 6/28/22, 1.13 on 10/28/22 and 2.0 on 3/15/23.

Q4: **I missed the 4/14 5PM PST deadline, is there any option to have an extension?**

A: **No, in order to meet our 4/28 goal, we must hold 4/14 as our deadline and will not accept any requests after the fact. We are over communicating the timelines and process with the community to avoid such issues.**

Q5: Where should I double check to see if my issue is in the cherry pick tracker?

A: [[v2.0.1] Release Tracker · Issue #97272 · pytorch/pytorch · GitHub](https://github.com/pytorch/pytorch/issues/97272)

Q6: Where can I find the Release Compatibility Matrix for PyTorch?

A: [pytorch/RELEASE.md at master · pytorch/pytorch · GitHub](https://github.com/pytorch/pytorch/blob/master/RELEASE.md#release-compatibility-matrix)

Please contact OSS Releng team members if you have any questions/comments. Again we appreciate everyone’s time and commitment to the community, PyTorch and 2.0 and 2.01 releases!


Please refer to this post for more details: https://dev-discuss.pytorch.org/t/pytorch-release-2-0-1-important-information/1176

atalman commented:
removing from milestones since not included in the release
